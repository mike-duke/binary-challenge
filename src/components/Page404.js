import React from 'react';

export const Page404 = (props) => {
  console.log(props)
  return (
    <div className='four-oh-four'>
      <h1>This page does not exist</h1>
      <a href="/">Return Home</a>
      <img src="https://media.giphy.com/media/uQkKavfX6TER2/giphy.gif" alt="Kirby hacking"/>
    </div>
  )
}