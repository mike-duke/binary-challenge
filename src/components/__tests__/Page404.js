import React from 'react';
import { shallow } from 'enzyme';
import { Page404 } from '../Page404';

describe('About', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Page404 />);

    expect(wrapper).toMatchSnapshot();
  });
});