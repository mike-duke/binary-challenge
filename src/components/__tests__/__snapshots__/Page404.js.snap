// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page404 />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          This page does not exist
        </h1>,
        <a
          href="/"
        >
          Return Home
        </a>,
        <img
          alt="Kirby hacking"
          src="https://media.giphy.com/media/uQkKavfX6TER2/giphy.gif"
        />,
      ],
      "className": "four-oh-four",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This page does not exist",
        },
        "ref": null,
        "rendered": "This page does not exist",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Return Home",
          "href": "/",
        },
        "ref": null,
        "rendered": "Return Home",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Kirby hacking",
          "src": "https://media.giphy.com/media/uQkKavfX6TER2/giphy.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            This page does not exist
          </h1>,
          <a
            href="/"
          >
            Return Home
          </a>,
          <img
            alt="Kirby hacking"
            src="https://media.giphy.com/media/uQkKavfX6TER2/giphy.gif"
          />,
        ],
        "className": "four-oh-four",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This page does not exist",
          },
          "ref": null,
          "rendered": "This page does not exist",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return Home",
            "href": "/",
          },
          "ref": null,
          "rendered": "Return Home",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Kirby hacking",
            "src": "https://media.giphy.com/media/uQkKavfX6TER2/giphy.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
